crates/esthri-cli/tests/integration/http_server.rs

```diff
@@ -373,11 +386,18 @@ fn test_fetch_archive_with_compressed_files() {
     let filename = esthri_test::test_data("index.html");
     let s3client = esthri_test::get_s3client();
     let s3_key = format!("{}/{}", prefix, "index_compressed.html");
-    let res = blocking::upload_compressed(
+    let opts = GenericOptParamsBuilder::default().build().unwrap();
+    let opts_compressed = GenericOptParamsBuilder::default()
```

Comment:
nit: I think I am more of a fan of this pattern where you can access the builder via a `::builder()` function on the original struct.

```rust
let opts = GenericOptParams::builder()
    .transparent_compression(true)
    .build();
```